### PR TITLE
Pass original requestor around during forwards

### PR DESCRIPTION
**Pull request by stgraber**
This is another pretty mechanical change related to requestors and their use in the lifecycle API.

With this, we're now passing additional HTTP headers when forwarding a request to another cluster member, those headers include the original username, protocol and address and are then pulled by the daemon logic on the target server and used to set the relevant context.

I've already added what's needed for the address in anticipation of #8856 but it's currently not directly used.

### CODE REVIEW

File: lxd/daemon.go [hunk 1]
```diff
@@ -527,6 +527,14 @@ func (d *Daemon) createCmd(restAPI *mux.Router, version string, c APIEndpoint) {
 			ctx = context.WithValue(ctx, "protocol", protocol)
 			ctx = context.WithValue(ctx, "access", userAccess)
 
+			// Add forwarded requestor data.
+			if protocol == "cluster" {
+				// Add authentication/authorization context data.
+				ctx = context.WithValue(ctx, "forwarded_address", r.Header.Get("X-LXD-cluster-address"))
```
Comment (tomponline) [inline lxd/daemon.go:533]: @stgraber please can we have constants defined for these so we can easily see where they are stored and used?

Comment (tomponline) [reply lxd/daemon.go:533]: And it probably makes sense to have constants for these to:

```go
			ctx := context.WithValue(r.Context(), "username", username)
			ctx = context.WithValue(ctx, "protocol", protocol)
			ctx = context.WithValue(ctx, "access", userAccess)
```

File: lxd/operations/operations.go [hunk 2]
```diff
@@ -183,14 +183,26 @@ func OperationCreate(s *state.State, projectName string, opClass OperationClass,
 		ctx := r.Context()
 		requestor := api.EventLifecycleRequestor{}
 
-		username, ok := ctx.Value("username").(string)
+		// Normal requestor.
+		val, ok := ctx.Value("username").(string)
```
Comment (tomponline) [inline lxd/operations/operations.go:187]: and using the constants here to tightly couple them together.

File: lxd/cluster/connect.go [hunk 3]
```diff
@@ -64,6 +66,28 @@ func Connect(address string, networkCert *shared.CertInfo, serverCert *shared.Ce
 		args.UserAgent = request.UserAgentNotifier
 	}
 
+	if r != nil {
+		proxy := func(req *http.Request) (*url.URL, error) {
+			ctx := r.Context()
+
+			val, ok := ctx.Value("username").(string)
```
Comment (tomponline) [inline lxd/cluster/connect.go:73]: constant use here too.

Also I wouldn't be adversed to having a separate set of constants for our HTTP headers in one place so we can see all the ones we use. Although that is probably outside of the scope of this change, and its a lot easier to search the code base for them.

**Comment by stgraber** [comment]
@tomponline that should do the trick

**Review by tomponline — APPROVED** [review]
Excellent thanks!

**Comment by stgraber** [comment]
@tomponline green enough? :)

**Pull request merged**